src/libraries/System.Formats.Asn1/src/System/Formats/Asn1/AsnWriter.cs

```diff
@@ -621,6 +621,12 @@ private static void SortContents(byte[] buffer, int start, int end)
 
             Debug.Assert(pos == end);
 
+            // If there is only one thing in the contents, we have no sorting work to do and it's already sorted.
+            if (positions.Count == 1)
```

Comment:
Count can't be 0?

Reply:
Count can only be 0 if the data set is empty, which was an earlier exit.